lib/formContext.ts

```diff
@@ -867,3 +869,142 @@ export const decrementNextActionChoiceIds = (groups: GroupsType, choiceId: strin
 
   return updatedGroups;
 };
+
+// TODO: in followup PR - tracked in #4407
+// 1) move Review.tsx's getReviewItemElements() to this file and refactor to be more generic
+// 2) update this function to use the above
+// 3) add unit tests where it makes sense
+export const getFormElementsFromGroups = ({
+  form,
+  values,
+  groupHistory,
+  groups,
+  matchedIds,
+  language, // TODO: will be refactored out when the above TODO is done
+}: {
+  form: FormProperties;
+  values: FormValues;
+  groupHistory: string[];
+  groups: GroupsType | object | undefined;
+  matchedIds: string[];
+  language: string;
+}) => {
+  return groupHistory
+    .map((groupId) => {
+      if (!groups) return [];
+      const group: Group = groups[groupId as keyof typeof groups] || {};
+      const groupElements = getReviewItemElements(
```

Comment:
If group is {} can we avoid the getReviewElements call --- we know it doesn't have elements